lib/tar_gz.ml

```diff
@@ -121,21 +100,26 @@ let really_read_through_gz decoder len =
 
 type error = [ `Fatal of Tar.error | `Eof | `Gz of string ]
 
-let seek_through_gz : decoder -> int -> (unit, [> error ], _) Tar.t = fun state len ->
+let seek_through_gz
+  : decoder -> int -> (unit, [> error ], _) Tar.t
+  = fun state len ->
   let open Tar in
   let* _buf = really_read_through_gz state len in
   Tar.return (Ok ())
 
-let gzipped t =
-  let rec go : type a. decoder -> (a, [> error ] as 'err, 't) Tar.t -> (a, 'err, 't) Tar.t = fun decoder -> function
+let in_gzipped t =
+  let rec go
+    : type a. decoder -> (a, [> error ] as 'err, 't) Tar.t -> (a, 'err, 't) Tar.t
+    = fun decoder -> function
     | Tar.Really_read len ->
       really_read_through_gz decoder len
     | Tar.Read _len -> assert false (* XXX(dinosaure): actually does not emit [Tar.Read]. *)
     | Tar.Seek len -> seek_through_gz decoder len
     | Tar.Return _ as ret -> ret
     | Tar.Bind (x, f) ->
       Tar.Bind (go decoder x, (fun x -> go decoder (f x)))
-    | Tar.High _ as high -> high in
+    | Tar.High _ as high -> high
+    | Tar.Write _ -> assert false in
```

Comment:
Would it make sense to leave `Tar.Write` untouched as is?

Reply:
You are right, we probably just keep it and remove an `assert false` :+1:.